optaplanner-examples/src/main/java/org/optaplanner/examples/tennis/optional/score/TennisConstraintProvider.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.tennis.optional.score;
+
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.core.api.score.stream.Joiners;
+import org.optaplanner.examples.tennis.domain.TeamAssignment;
+import org.optaplanner.examples.tennis.domain.UnavailabilityPenalty;
+
+import static org.optaplanner.core.api.score.stream.ConstraintCollectors.loadBalanceByCount;
+
+public class TennisConstraintProvider implements ConstraintProvider {
+
+    @Override
+    public Constraint[] defineConstraints(ConstraintFactory constraintFactory) {
+        return new Constraint[] {
+                oneAssignmentPerDatePerTeam(constraintFactory),
+                unavailabilityPenalty(constraintFactory)
+        };
+    }
+
+    // ############################################################################
+    // Hard constraints
+    // ############################################################################
+
+    private Constraint oneAssignmentPerDatePerTeam(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(TeamAssignment.class)
+                .ifExistsOther(TeamAssignment.class,
+                        Joiners.equal(TeamAssignment::getTeam),
+                        Joiners.equal(TeamAssignment::getDay),
+                        Joiners.lessThan(TeamAssignment::getId),
```

Comment:
Doesn't the "Other" in ifExistsOther already do that? @triceo wdyt?
Well not really: other means unequal, no uniqueness of pair, right?
This is confusing. How often do we want to use "other" without "uniqueness of pair"? I 'd argue never?

Reply:
I was actually considering that when I was designing the API, we even discussed that. We've decided to stick to what `join()` does. With `join()`, you also have to handle uniqueness manually (or use `fromUniquePair()`).

Reply:
Yes, I remember.
Seeing this example, I started to doubt our decision (a bit late as 7.33 is out).
We can leave it as is for now, and keep it in the back of our head.

But this example should drop the "Other" because it already has a "lessThan(id)"